CONTRIBUTING.md

```diff
@@ -71,6 +71,69 @@ $ git rebase master
 
 > Now your `feature/my-new-feature` branch is up-to-date with all the code in `upstream/master`.
 
+Running E2E Tests locally before commits
+----------------------------------------
+E2E Tests can be run against the Custom Login, Okta-Hosted Login and Resource servers
+
+Follow the steps below to run the tests locally:
+
+```bash
+# At project root
+composer install
+```
+To test the samples you will need the following configured in your developer org:
+
+* [A Web application](/okta-hosted-login#prerequisites)
+* [A SPA application](https://github.com/okta/samples-js-angular/tree/master/okta-hosted-login#prerequisites)
+* A test user account with a known username and password. Note that the USERNAME should be of the form "username@email.com"
+
+Once you have those resources setup, export their details as the following environment variables:
+
+```bash
+export ISSUER=https://{yourOktaDomain}.com/oauth2/default
+export CLIENT_ID={yourWebAppClientId}
+export CLIENT_SECRET={yourWebAppClientSecret}
+export SPA_CLIENT_ID={yourSpaAppClientId}
+export USERNAME={userName}
+export PASSWORD={password}
+```
+
+For Windows, please set the following environment variables:
+- `ISSUER`
+- `CLIENT_ID`
+- `CLIENT_SECRET`
+- `SPA_CLIENT_ID`
+- `USER_NAME`
+- `PASSWORD`
+
+As an alternative you can provide the environment variables in a file named `testenv` in the root folder.
+
+For example:
+
+```
+ISSUER=https://dev-12345.oktapreview.com/oauth2/default
+CLIENT_ID=webclient123
+CLIENT_SECRET=websecret123
+SPA_CLIENT_ID=spaclient123
+USERNAME=myuser@example.com
+PASSWORD=mypassword
```

Comment:
Lets make sure the values we supply here match the above `export` commands. We should be throwing errors immediately if `{yourOktaDomain}` is the set `ISSUER`, while `dev-12345` will not throw.

Reply:
This is not a file that will be checked in. The user has to create a `testenv` file and provide the values manually. I don't think there is a chance that users will provide `{yourOktaDomain}` in it.